diff --git a/pygments/lexers/templates.py b/pygments/lexers/templates.py
index 4084fbd50..bb3c633db 100644
--- a/pygments/lexers/templates.py
+++ b/pygments/lexers/templates.py
@@ -1445,7 +1445,7 @@ class EvoqueLexer(RegexLexer):
     aliases = ['evoque']
     filenames = ['*.evoque']
     mimetypes = ['application/x-evoque']
-    url = 'https://gizmojo.org/templating'
+    url = ''
     version_added = '1.1'
 
     flags = re.DOTALL
@@ -1503,7 +1503,7 @@ class EvoqueHtmlLexer(DelegatingLexer):
     aliases = ['html+evoque']
     filenames = ['*.html']
     mimetypes = ['text/html+evoque']
-    url = 'https://gizmojo.org/templating'
+    url = ''
     version_added = '1.1'
 
     def __init__(self, **options):
@@ -1522,7 +1522,7 @@ class EvoqueXmlLexer(DelegatingLexer):
     aliases = ['xml+evoque']
     filenames = ['*.xml']
     mimetypes = ['application/xml+evoque']
-    url = 'https://gizmojo.org/templating'
+    url = ''
     version_added = '1.1'
 
     def __init__(self, **options):
